src/controls.ts

```diff
@@ -5,15 +5,49 @@ SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
 Please see LICENSE in the repository root for full details.
 */
 
-import { Subject } from "rxjs";
+import { BehaviorSubject, Subject } from "rxjs";
 
 export interface Controls {
-  canEnterPip: () => boolean;
-  enablePip: () => void;
-  disablePip: () => void;
+  canEnterPip(): boolean;
+  enablePip(): void;
+  disablePip(): void;
+  setAvailableOutputDevices(devices: OutputDevice[]): void;
+  setOutputDevice(id: string): void;
+  onOutputDeviceSelect?: (id: string) => void;
+  setOutputEnabled(enabled: boolean): void;
+  showNativeOutputDevicePicker?: () => void;
 }
 
+export interface OutputDevice {
+  id: string;
+  name: string;
+  forEarpiece?: boolean;
+  isEarpiece?: boolean;
+  isSpeaker?: boolean;
+  isExternalHeadset?: boolean;
+}
+
+/**
+ * If pipMode is enabled, EC will render a adapted call view layout.
+ */
 export const setPipEnabled$ = new Subject<boolean>();
+// BehaviorSubject since the client might set this before we have subscribed (GroupCallView still in "loading" state)
+// We want the that has been set during loading to be be available immediately once loaded.
+export const setAvailableOutputDevices$ = new BehaviorSubject<OutputDevice[]>(
```

Comment:
As long as these are BehaviorSubjects (representing stateful values rather than events), I think it makes sense to name these like values:

```suggestion
export const availableOutputDevices$ = new BehaviorSubject<OutputDevice[]>(
```